ssl/statem/statem_lib.c

```diff
@@ -1004,6 +1005,73 @@ int ssl_choose_server_version(SSL *s, CLIENTHELLO_MSG *hello)
         break;
     }
 
+    suppversions = tls_get_extension_by_type(hello->pre_proc_exts,
+                                             hello->num_extensions,
+                                             TLSEXT_TYPE_supported_versions);
+
+    /*
+     * TODO(TLS1.3): We only look at this if our max protocol version is TLS1.3
+     * or above. Should we allow it for lower versions too?
```

Comment:
The way I look at this is that the if this extension is send, the client is using it to send it's list of supported protocols. If the server doesn't enable TLS 1.3 support, we can still look at what the clients wants.

In your case hello.version == TLS1.2 and TLS 1.1 in the extension, it looks like it's a client that doesn't support TLS 1.2 or 1.3 and we should look at it like one that supports TLS 1.1. We might decide never to send it (and you just removed the TODO), but other clients might decide to always send this extension.

We should either ignore the hello.version in case this extension is present, or insist that it's set to TLS 1.2.